### PR TITLE
[#5977] Update projects to .Net 6 - Functional Test pipelines

### LINKED ISSUE #5977: Update projects to .Net 6
## Summary

Update integration projects, test projects and web app projects to include .Net 6.

As part of this issue we will add and test support for Net 6 in the dotnet azure SDK.

- [ ] Add `net6.0` to the TargetFrameworks list to the following libraries:
Microsoft.Bot.Builder.Integration.AspNet.Core
Microsoft.Bot.Builder.Integration.ApplicationInsights.Core
Microsoft.Bot.Builder.Dialogs.Adaptive.Runtime
- [ ] Add `net6.0` to the TargetFrameworks list of the unit test projects (unit tests should run on both versions)
- [ ] Update the web tests projects in the solution to target Net 6 (we don't need multi targetting for the testing projects we have in the Tests folder.
- [ ] Make required changes to the build pipelines if needed

### Reference articles

- [What's new in .NET 6](https://docs.microsoft.com/en-us/dotnet/core/whats-new/dotnet-6)
- [Breaking changes in .NET 6](https://docs.microsoft.com/en-us/dotnet/core/compatibility/6.0?toc=/dotnet/fundamentals/toc.json&bc=/dotnet/breadcrumb/toc.json)

### PR DESCRIPTION
**Pull request by ceciliaavila**
Fixes #5977

## Description
This PR updates the FunctionalTests pipelines to target .NET6.

**_Note: The PR depends on PR #6332 with the changes in the test projects_.**

## Specific Changes
- Updated the _vmImages_ from windows-2019 to windows-2022.
- Changed framework from netcoreapp3.1 to net6.0 in `build/yaml/functional-test-setup-steps.yml`.

## Testing
Here we can see the pipelines working after the changes.
image

### CODE REVIEW

**Comment by ceciliaavila** [comment]
Hi @BruceHaley, in this PR we updated the FT pipelines to target .NET6. It'd be great if you could take a look at the changes.
Thanks!

File: build/yaml/botbuilder-dotnet-ci-facebook-test-secondary.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 name: $(Build.BuildId)
 
 pool:
-  vmImage: $[ coalesce( variables['VMImage'], 'windows-2019' ) ] # or 'windows-latest' or 'vs2017-win2016'
+  vmImage: $[ coalesce( variables['VMImage'], 'windows-2022' ) ] # or 'windows-latest' or 'vs2017-win2016'
```
Comment (BruceHaley) [inline build/yaml/botbuilder-dotnet-ci-facebook-test-secondary.yml:9]: Please drop 'vs2017-win2016' from the comment, as it is no longer supported.

File: build/yaml/botbuilder-dotnet-ci-facebook-test.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@
 name: $(Build.BuildId)
 
 pool:
-  vmImage: $[ coalesce( variables['VMImage'], 'windows-2019' ) ] # or 'windows-latest' or 'vs2017-win2016'
+  vmImage: $[ coalesce( variables['VMImage'], 'windows-2022' ) ] # or 'windows-latest' or 'vs2017-win2016'
```
Comment (BruceHaley) [inline build/yaml/botbuilder-dotnet-ci-facebook-test.yml:9]: Please drop 'vs2017-win2016' from the comment, as it is no longer supported.

File: build/yaml/botbuilder-dotnet-ci-slack-test.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@
 name: $(Build.BuildId)
 
 pool:
-  vmImage: $[ coalesce( variables['VMImage'], 'windows-2019' ) ] # or 'windows-latest' or 'vs2017-win2016'
+  vmImage: $[ coalesce( variables['VMImage'], 'windows-2022' ) ] # or 'windows-latest' or 'vs2017-win2016'
```
Comment (BruceHaley) [inline build/yaml/botbuilder-dotnet-ci-slack-test.yml:9]: Please drop 'vs2017-win2016' from the comment, as it is no longer supported.

File: build/yaml/botbuilder-dotnet-ci-twilio-test.yml [hunk 4]
```diff
@@ -6,7 +6,7 @@
 name: $(Build.BuildId)
 
 pool:
-  vmImage: $[ coalesce( variables['VMImage'], 'windows-2019' ) ] # or 'windows-latest' or 'vs2017-win2016'
+  vmImage: $[ coalesce( variables['VMImage'], 'windows-2022' ) ] # or 'windows-latest' or 'vs2017-win2016'
```
Comment (BruceHaley) [inline build/yaml/botbuilder-dotnet-ci-twilio-test.yml:9]: Please drop 'vs2017-win2016' from the comment, as it is no longer supported.

File: build/yaml/botbuilder-dotnet-ci-webex-test.yml [hunk 5]
```diff
@@ -6,7 +6,7 @@
 name: $(Build.BuildId)
 
 pool:
-  vmImage: $[ coalesce( variables['VMImage'], 'windows-2019' ) ] # or 'windows-latest' or 'vs2017-win2016'
+  vmImage: $[ coalesce( variables['VMImage'], 'windows-2022' ) ] # or 'windows-latest' or 'vs2017-win2016'
```
Comment (BruceHaley) [inline build/yaml/botbuilder-dotnet-ci-webex-test.yml:9]: Please drop 'vs2017-win2016' from the comment, as it is no longer supported.

File: build/yaml/botbuilder-dotnet-functional-test-windows.yml [hunk 6]
```diff
@@ -6,7 +6,7 @@
 name: $(Build.BuildId)
 
 pool:
-  vmImage: $[ coalesce( variables['VMImage'], 'windows-2019' ) ] # or 'windows-latest' or 'vs2017-win2016'
+  vmImage: $[ coalesce( variables['VMImage'], 'windows-2022' ) ] # or 'windows-latest' or 'vs2017-win2016'
```
Comment (BruceHaley) [inline build/yaml/botbuilder-dotnet-functional-test-windows.yml:9]: ...and so forth.

**Review by BruceHaley — APPROVED** [review]
Looks good. The comments could use fixing as noted.

File: build/yaml/botbuilder-dotnet-ci-facebook-test-secondary.yml (hunk 1 above)
Comment (ceciliaavila) [reply build/yaml/botbuilder-dotnet-ci-facebook-test-secondary.yml]: Comment updated. Thanks!

File: build/yaml/botbuilder-dotnet-ci-slack-test.yml (hunk 3 above)
Comment (ceciliaavila) [reply build/yaml/botbuilder-dotnet-ci-slack-test.yml]: Comment updated. Thanks!

File: build/yaml/botbuilder-dotnet-ci-twilio-test.yml (hunk 4 above)
Comment (ceciliaavila) [reply build/yaml/botbuilder-dotnet-ci-twilio-test.yml]: Comment updated. Thanks!

File: build/yaml/botbuilder-dotnet-ci-webex-test.yml (hunk 5 above)
Comment (ceciliaavila) [reply build/yaml/botbuilder-dotnet-ci-webex-test.yml]: Comment updated. Thanks!

File: build/yaml/botbuilder-dotnet-functional-test-windows.yml (hunk 6 above)
Comment (ceciliaavila) [reply build/yaml/botbuilder-dotnet-functional-test-windows.yml]: All comments updated. Thanks!

**Review by gabog — APPROVED** [review]

**Pull request merged**